discord/user/member.py

```diff
@@ -1,27 +1,44 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from datetime import datetime
+from typing import List, Optional, Any
 
-from .user import User
+from pydantic import validator
+import arrow
 
-if TYPE_CHECKING:
-    from ..guild import Guild
-    from ..role import Role
+from .user import User
+from ..asset import Asset
+from ..types.snowflake import Snowflake
 
 
-class Member(User):
-    _top_role: Role
-    _roles: set[Role]
-    _guild: Guild
+def validate_dt(val):
+    if val is None:
+        return val
 
-    @property
-    def top_role(self):
-        return self._top_role
+    return arrow.get(str(val))
 
-    @property
-    def roles(self):
-        return self._roles
 
-    @property
-    def guild(self):
-        return self._guild
+class Member(User):
+    nick: Optional[str] = None
```

Comment:
Again, using these are against the read-only idea of some attributes

Reply:
pydantic models are immutable.